### PR TITLE
[DEV-1386] add exchangeTx entity

### PR DESCRIPTION
**Pull request by Yannick1712**
#### Pre-Release

- [ ] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [ ] Check for linter errors (in PR)
- [ ] Test basic user operations
  - Login/logout
  - Registration
  - Update user data
  - Create/delete payment routes

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

**Comment by Yannick1712** [comment]
Update Env + deactivate blackSquad Mails before merging in master!

File: src/integration/exchange/exchange.module.ts
Comment (davidleomay) [inline src/integration/exchange/exchange.module.ts:1]: Need to add the repository to the module?

File: src/integration/exchange/entities/exchange-tx.entity.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+  @Column({ length: 256, nullable: true })
+  side: string;
+
+  @Column({ length: 256, nullable: true })
+  takeOrMaker: string;
+
+  @Column({ type: 'float', nullable: true })
+  fees: number;
+}
```
Comment (davidleomay) [inline src/integration/exchange/entities/exchange-tx.entity.ts:128]: Missing migration

File: src/integration/exchange/entities/exchange-tx.entity.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  aclass: string;
+
+  @Column({ length: 256, nullable: true })
+  asset: string;
+
+  @Column({ length: 256, nullable: true })
+  refid: string;
+
+  @Column({ length: 256, nullable: true })
+  txid: string;
```
Comment (davidleomay) [inline src/integration/exchange/entities/exchange-tx.entity.ts:34]: txId

File: src/integration/exchange/entities/exchange-tx.entity.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  method: string;
+
+  @Column({ length: 256, nullable: true })
+  aclass: string;
+
+  @Column({ length: 256, nullable: true })
+  asset: string;
+
+  @Column({ length: 256, nullable: true })
+  refid: string;
```
Comment (davidleomay) [inline src/integration/exchange/entities/exchange-tx.entity.ts:31]: refId

File: src/integration/exchange/entities/exchange-tx.entity.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  dateTime: Date;
+
+  @Column({ type: 'float' })
+  cost: number;
+
+  @Column({ length: 256, nullable: true })
+  order: string;
+
+  @Column({ length: 256, nullable: true })
+  orderTxid: string;
```
Comment (davidleomay) [inline src/integration/exchange/entities/exchange-tx.entity.ts:88]: orderTxId

File: src/integration/exchange/entities/exchange-tx.entity.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+  @Column({ length: 256, nullable: true })
+  posTxid: string;
```
Comment (davidleomay) [inline src/integration/exchange/entities/exchange-tx.entity.ts:91]: posTxId

File: src/integration/exchange/entities/exchange-tx.entity.ts [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+  @Column({ length: 256, nullable: true })
+  pair: string;
+
+  @Column({ length: 256, nullable: true })
+  ordertype: string;
```
Comment (davidleomay) [inline src/integration/exchange/entities/exchange-tx.entity.ts:97]: orderType

**Review by davidleomay — APPROVED** [review]
Add migration

**Review by davidleomay — APPROVED** [review]
Add migration

**Pull request merged**